src/main/resources/lib/enonic/react4xp/pageContributions/buildPageContributions.ts

```diff
@@ -1,14 +1,157 @@
 import type {
 	OneOrMore,
 	PageContributions,
-	React4xp as React4xpNamespace
-} from '../../../../index.d';
-
+	React4xp as React4xpNamespace,
+	// Request
+} from '/types';
+import type {Content} from '/lib/xp/portal';
 
+import {includes as arrayIncludes} from '@enonic/js-utils/array/includes';
 import endsWith from '@enonic/js-utils/string/endsWith';
+import { toStr } from '@enonic/js-utils/value/toStr';
 import {getClientUrl} from '/lib/enonic/react4xp/asset/client/getClientUrl';
-import {getGlobalsUrls} from '../asset/globals/getGlobalsUrls';
+import {getGlobalsUrls} from '/lib/enonic/react4xp/asset/globals/getGlobalsUrls';
 import {getComponentChunkUrls} from '/lib/enonic/react4xp/dependencies/getComponentChunkUrls';
+import {
+	getResource,
+	readText,
+} from '/lib/xp/io';
+import {getContent} from '/lib/xp/portal';
+
+
+const RESOURCE_KEY = Java.type('com.enonic.xp.resource.ResourceKey');
+
+
+function anEntryFoundInAReact4xpApplication({
+	applicationKey,
+	appNameToEntries,
+	entries,
+}: {
+	applicationKey: string,
+	appNameToEntries: Record<string,string[]>
+	entries: OneOrMore<React4xpNamespace.EntryName>
+}) {
+	if (!appNameToEntries[applicationKey]) {
+		const filePath = '/r4xAssets/entries.json';
+		const resourcePath = `${applicationKey}:${filePath}`;
+		const resource = getResource(RESOURCE_KEY.from(resourcePath));
```

Comment:
getResource supports string, doesn't it?

Reply:
Yes, but not from another app.

Reply:
sure about that? path starting from `getResource('/r4xAssets/entries.json')` should do the trick. Also you can `require` the file instead of using resource magic.

Reply:
Remember this is in a multi app setting. Each app has it's own '/r4xAssets/entries.json' file. I'm reading a file inside another app.